src/silo_api/database_directory_watcher.cpp

```diff
@@ -137,6 +137,11 @@ void silo_api::DatabaseDirectoryWatcher::checkDirectoryForData(Poco::Timer& /*ti
    try {
       database_mutex.setDatabase(silo::Database::loadDatabaseState(most_recent_database_state->first
```

Comment:
Is there a timeout on this? When the incident happened, is this where it got stuck indefinitely? Just to help me understand what happened.